app/models/spree/product.rb

```diff
@@ -29,6 +29,10 @@ class Product < ApplicationRecord
 
     acts_as_paranoid
 
+    after_create :ensure_standard_variant
+    around_destroy :destruction
```

Comment:
@mkllnk , do you know if the `transaction` block is necessary? It was added in 836a08606c1f2a57ce5c450b665b3cef128703ce, but the product model specs all succeed without.

If not, we could make it into a couple of `before_destroy` callbacks instead:

```suggestion
    before_destroy :touch_distributors
    before_destroy :destroy_exchange_variants
```

Reply:
> do you know if the transaction block is necessary?

Hm, good question. In general I would always say that a transaction is a good idea to not partially delete data. But in this case we

* touch distributors which isn't harmful at all except for lower performance,
* remove the product from all order cycles which wouldn't be the end of the world when your attempt to delete a product fails. You probably wanted that to happen anyway.

Still, when I see an error message I would assume that the whole action failed instead of some data being removed. I don't mind if we change it in this case but I personally wouldn't bother.

Reply:
Thanks, I have to admit for some reason I was thinking that everything happens in an implicit transaction anyway, but then quickly realised I haven't touched transactions for a while and can't remember how they all work.

It does make sense to specify a transaction here; I don't know what I was thinking. Probably just got excited about cleaning up ;)